src/node/board.js

```diff
@@ -1,28 +1,61 @@
-const BoardConstruct = require('./boardConstructor');
-const BoardPrinter = require('./boardPrinter.js');
-const Point = require('./point.js');
+const constants = require('./constants.js');
+const Tile = require('./tile.js');
 
-function Board(size) {
+function Board() {
   const board = {
-    size,
-    matrix: BoardConstruct(size),
-    placePiece(point, symbol) {
-      board.matrix[point.y][point.x].setSymbol(symbol);
+    matrix: BoardConstruct(),
+    placeSymbol: (point, symbol) => {
+      board.matrix[point.x][point.y].setSymbol(symbol);
+      board.matrix[point.x][point.y].setOccupiedToTrue();
     },
-    print: () => {
+    tileIsOccupied: point => {
+      return board.matrix[point.x][point.y].isOccupied;
+    },
+    getSymbol: point => {
+      return board.matrix[point.x][point.y].symbol;
+    },
+    print() {
       BoardPrinter(board);
     },
   };
 
   return board;
 }
 
-//TEST
-const board = Board(3);
-let a = Point(1, 2);
-board.placePiece(a, 'X');
-a = Point(0, 0);
-board.placePiece(a, 'O');
-board.print();
+function BoardConstruct() {
+  const matrixOfTiles = [];
+
+  for (let i = 0; i < constants.boardSize; i += 1) {
+    const rowOfTiles = [];
+    for (let j = 0; j < constants.boardSize; j += 1) {
+      rowOfTiles.push(Tile());
+    }
+    matrixOfTiles.push(rowOfTiles);
+  }
+  return matrixOfTiles;
+}
+
+function BoardPrinter(board) {
+  let rowIndex = 0;
+  let boardString = '  0   1   2\n';
+
+  board.matrix.forEach(row => {
+    boardString += `${rowIndex} `;
+    row.forEach(tile => {
+      boardString += `${tile.symbol}`;
+      if (tile !== row[row.length - 1]) {
+        boardString += ' | ';
+      }
+    });
+
+    if (rowIndex !== constants.boardSize - 1) {
+      boardString += '\n  - | - | -\n';
+    }
+    rowIndex++;
```

Comment:
`forEach` doesn't iterate only over the items of the array! [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach)
![image](https://user-images.githubusercontent.com/11878589/84039284-6b065180-a977-11ea-8855-9abdc70f9409.png)
Here is the function signature! Don't let the `[` scare you because it just means that this parameter is optional.
This signature for array.forEach means that it reaceives 2 arguments, a `callback` and an optional `this`:
``` js
function callback() {}
[].forEach(callback, this)
```
The callback parameter, can receive 3 arguments: the current value from iteration, the index of the iteration and the  whole array.
 ``` js
const array = [1,2,3]
function callback(item, index, arr) {
  console.log(arr === array) // this is true
}
array.forEach(callback, this)
```

Reply:
This way you don't need to manually manage the index yourself, you could just use the index the callback gives you (and it's the recommended way to have functions only operating with things from the arguments instead of using variables outside the function)